packages/url_launcher/url_launcher_android/example/integration_test/url_launcher_test.dart

```diff
@@ -20,4 +23,44 @@ void main() {
     // sms:, tel:, and mailto: links may not be openable on every device, so
     // aren't tested here.
   });
+
+  testWidgets('launch and close', (WidgetTester _) async {
+    final UrlLauncherPlatform launcher = UrlLauncherPlatform.instance;
+
+    // Setup fake http server.
+    final HttpServer server = await HttpServer.bind(InternetAddress.anyIPv4, 0);
+    unawaited(server.forEach((HttpRequest request) {
+      if (request.uri.path == '/hello.txt') {
+        request.response.writeln('Hello, world.');
+      } else {
+        fail('unexpected request: ${request.method} ${request.uri}');
+      }
+      request.response.close();
+    }));
+    // Https to avoid cleartext warning on android.
+    final String prefixUrl = 'https://${server.address.address}:${server.port}';
+    final String primaryUrl = '$prefixUrl/hello.txt';
+
+    // Launch a url then close.
+    expect(
+        await launcher.launch(primaryUrl,
+            useSafariVC: true,
+            useWebView: true,
+            enableJavaScript: false,
+            enableDomStorage: false,
+            universalLinksOnly: false,
+            headers: <String, String>{'my_header_key': 'my_header_value'}),
```

Comment:
The headers map can be empty since they aren't related to the test.

Reply:
done